cpp/daal/src/algorithms/dtrees/forest/classification/oneapi/df_classification_train_hist_oneapi_impl.i

```diff
@@ -280,9 +282,15 @@ services::Status ClassificationTrainBatchKernelOneAPI<algorithmFPType, hist>::co
             //_maxLocalHistograms/2 (128) showed better performance than _maxLocalHistograms need to investigate
             int reduceLocalSize = 16; // add logic for its adjustment
 
-            size_t nHistBins    = nSelectedFeatures * _nMaxBinsAmongFtrs;
+            DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, nSelectedFeatures, _nMaxBinsAmongFtrs);
+            size_t nHistBins = nSelectedFeatures * _nMaxBinsAmongFtrs;
+
+            DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, nHistBins, _nMaxBinsAmongFtrs);
             size_t partHistSize = nHistBins * _nClasses;
 
+            DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, nGroupNodes, partHistSize);
+            DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, nGroupNodes * partHistSize, nPartialHistograms);
+
             auto partialHistograms = context.allocate(TypeIds::id<algorithmFPType>(), nGroupNodes * nPartialHistograms * partHistSize, &status);
```

Comment:
Separate status check for every allocation

Reply:
fixed